server/handlers.go

```diff
@@ -668,6 +668,44 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 			s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
 			return
 		}
+
+		tokenRef := storage.RefreshTokenRef{
+			ID:        refresh.ID,
+			CreatedAt: refresh.CreatedAt,
+			LastUsed:  refresh.LastUsed,
+		}
+
+		if session, err := s.storage.GetOfflineSession(refresh.Claims.UserID, refresh.ConnectorID); err != nil {
```

Comment:
This should be isolated to only when `err == storage.ErrNotFound`. Other errors need to be reported as errors.